### PR TITLE
interface for adding/removing drinks

### PR DESCRIPTION
**Pull request by dpopp07**
Created the interface for add/remove mutations for editing the catalog. The guts are missing but I thought I would make the PR now in case you wanted to see what the mutations would look like from your perspective.

Also, this could be changed. The primary thing I question is the payload. Right now, I'm just returning the drink object that was added/removed if successful. The other option is to return the whole catalog. Let me know what you think.

Todo:
- [x] Fill out resolvers/logic
- [x] Add tests
- [x] Change add/remove to take in arrays for batch processing

### CODE REVIEW

**Comment by dpopp07** [comment]
Tests have been added. Ready for review

File: bartop-api/src/api/catalog/catalog.graphql.js [hunk 1]
```diff
@@ -19,8 +20,63 @@ type ReplaceCatalogPayload {
...
+type AddDrinkToCatalogPayload {
+
+  # The Drink object added to the catalog
+  drink: Drink
+
+  # A flag that will be 'true' if the given drink is already present in the catalog, and 'null' otherwise.
+  unchanged: Boolean
+
+  # A flag that will be 'true' if the given drink ID does not exist in the database, and 'null' otherwise
+  invalidDrink: Boolean
```
Comment (jamesvclements) [inline bartop-api/src/api/catalog/catalog.graphql.js]: Would it make more sense to have `Payload` types always have an `errors` key, which can contain an array of the different `error`s that were caused by the input? It's hard to find what the exact best practice is but I found some jawns about it:

* https://stackoverflow.com/questions/32662437/graphql-and-form-validation-errors
* https://medium.com/@tarkus/validation-and-user-errors-in-graphql-mutations-39ca79cd00bf
* https://stackoverflow.com/questions/35713889/handling-errors-in-mutations

It just feels weird to have an exact key in the payload that is just for one particular error (an invalid drink) vs. all payloads can have `errors`, and the client would be responsible for looking at the `errors` array and handling appropriately

**Review by jamesvclements — CHANGES_REQUESTED** [review]

File: bartop-api/src/api/catalog/catalog.graphql.js [hunk 2]
```diff
@@ -19,8 +20,63 @@ type ReplaceCatalogPayload {
...
+  drinkId: ID!
+}
+
+# Return type of addDrinkToCatalog
+type AddDrinkToCatalogPayload {
+
+  # The Drink object added to the catalog
+  drink: Drink
+
+  # A flag that will be 'true' if the given drink is already present in the catalog, and 'null' otherwise.
```
Comment (jamesvclements) [inline bartop-api/src/api/catalog/catalog.graphql.js]: What's the use-case for this? 🤔

File: bartop-api/src/api/catalog/catalog.graphql.js [hunk 3]
```diff
@@ -19,8 +20,63 @@ type ReplaceCatalogPayload {
...
+  unchanged: Boolean
```
Comment (jamesvclements) [inline bartop-api/src/api/catalog/catalog.graphql.js]: What's the use-case for this? 🤔

**Comment by dpopp07** [comment]
Ready for review

**Comment by dpopp07** [comment]
A few things to point out:
1. I made a generic `Error` type for the GraphQL schema. That obviously can change but since I'm not 100% set on my error handling strategy yet I'm cool with that staying.
2. I discovered a bug (thankfully) that the `reql` queries I wrote will fail if the user doesn't already have a `catalog` object. Rather than implement some funky `reql` magic, I am just adding an empty catalog for every user. Effective and simple.
3. I discovered that if I throw an error in my code, it will not reach my error handler middleware. GraphQL catches it and sends a 500. Just something to know. I'm going to open an issue about error handling

**Review by jamesvclements — APPROVED** [review]
Hermosa

**Pull request merged**